src/app/services/views/ServiceDetailView.vue

```diff
@@ -39,125 +39,109 @@
 
         <EmptyBlock v-else-if="data === undefined" />
 
-        <DataSource
+        <TabsWidget
           v-else
-          v-slot="{ data: externalService, isLoading: isLoadingExternalService, error: externalServiceError }: ExternalServiceSource"
-          :should-make-request="data.serviceType === 'external'"
-          :src="`/meshes/${route.params.mesh}/external-services/${route.params.service}`"
+          :tabs="TABS.filter((tab) => {
+            if (tab.hash === '#overview') {
+              return true
+            } else {
+              return data.serviceInsight.serviceType !== 'external'
+            }
+          })"
```

Comment:
Does it make sense to inlide TABS here? I find it would be easier to read:

```typescript
:tabs="(() => {
  var r = [{
    hash: '#overview',
    title: t('services.routes.item.tabs.overview'),
  }]
  if (data.serviceInsights.serviceType !== 'external') {
    r.push({
      hash: '#dataPlaneProxies',
      title: t('services.routes.item.tabs.data_plane_proxies'),
    })
  }
  return r
})()"
```

Reply:
I really don’t like this style of inlining logic in the template. I would actually rather move the filtering I added here into the script and have something like this:

```ts
function getTabs(serviceInsight: ServiceInsight) {
  return [
    {
      hash: '#overview',
      title: t('services.routes.item.tabs.overview'),
    },
    {
      hash: '#dataPlaneProxies',
      title: t('services.routes.item.tabs.data_plane_proxies'),
    },
  ].filter((tab) => {
    if (tab.hash === '#overview') {
      return true
    } else {
      return serviceInsight.serviceType !== 'external'
    }
  })
}
```

There are cases when it’s just very convenient to inline a simple expression, but generally, it makes templates much harder to read for me.

Reply:
Abstract it as a function if you want. However, doing a filter rather than just building the right array is just making things harder to read.

Reply:
I’ve moved this logic into a function and turned into direct array construction here: https://github.com/kumahq/kuma-gui/pull/1345/commits/dd7c78ac21d3762bc64260656ec4a86792302371